eth/api.go

```diff
@@ -545,3 +545,27 @@ func (api *PrivateDebugAPI) getModifiedAccounts(startBlock, endBlock *types.Bloc
 	}
 	return dirty, nil
 }
+
+// GetBlockReceipts returns all transaction receipts of the specified block.
+func (api *PrivateDebugAPI) GetBlockReceipts(ctx context.Context, blockHash common.Hash) ([]map[string]interface{}, error) {
+	if block := api.eth.blockchain.GetBlockByHash(blockHash); block != nil {
+		if receipts := api.eth.blockchain.GetReceiptsByHash(blockHash); receipts != nil {
+			txs := block.Transactions()
+			if len(txs) != len(receipts) {
+				return nil, fmt.Errorf("txs length doesn't equal to receipts' length")
+			}
+			txReceipts := make([]map[string]interface{}, 0, len(txs))
+			blockNumber := block.Header().Number
+			for idx, receipt := range receipts {
+				tx := txs[idx]
+				fields, err := ethapi.ToTransactionReceipt(ctx, api.eth.APIBackend, tx, receipt, blockHash, tx.Hash(), blockNumber.Uint64(), uint64(idx))
```

Comment:
Not a strong opinion but block-related fields are being duplicated in every tx object, whereas they can be stored in a top-level object with potentially other fields, e.g. blockBloom:

`{
    hash: <blockhash>,
    number: <blocknumber>,
    bloom: <blockBloom>,
    receipts/txs: [tx0_receipt, tx1_receipt],
}`

Reply:
Thanks for your opinion, my concern is GetTransactionReceipt and GetBlockReceipts's response should be related(fields and []fields), if I changed GetBlockReceipts's return fields maybe cause some confusion?